components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1875,12 +1875,11 @@ where
     }
 
     fn on_ready_change_peer(&mut self, cp: ChangePeer) {
-        if cp.conf_change.get_node_id() == raft::INVALID_ID {
+        if cp.index == raft::INVALID_INDEX {
```

Comment:
When will it be `INVALID_INDEX`? I can't find it's set at any places.

Reply:
If apply failed, `Apply::default()` will be sent back, in which case `cp.index == raft::INVALID_INDEX == 0`

Reply:
Can you show me where it's constructed?

Reply:
https://github.com/tikv/tikv/blob/master/components/raftstore/src/store/fsm/apply.rs#L955